data/example_realization_config_w_bmi_c.json

```diff
@@ -0,0 +1,151 @@
+{
+    "global": {
+      "formulations": [
+        {
+          "name":"tshirt",
+          "params": {
+              "maxsmc": 0.439,
+              "wltsmc": 0.066,
+              "satdk": 0.00000338,
+              "satpsi": 0.355,
+              "slope": 1.0,
+              "scaled_distribution_fn_shape_parameter": 4.05,
+              "multiplier": 0.0,
+              "alpha_fc": 0.33,
+              "Klf": 0.01,
+              "Kn": 0.03,
+              "nash_n": 2,
+              "Cgw": 0.01,
+              "expon": 6.0,
+              "max_groundwater_storage_meters": 1.0,
+              "nash_storage": [
+                  0.0,
+                  0.0
+              ],
+              "soil_storage_percentage": 0.667,
+              "groundwater_storage_percentage": 0.5,
+              "timestep": 3600,
+              "giuh": {
+                  "giuh_path": "./test/data/giuh/GIUH.json",
+                  "crosswalk_path": "./data/crosswalk.json"
+              }
+            }
+          }
+        ],
+        "forcing": {
+            "file_pattern": ".*{{ID}}.*.csv",
+            "path": "./data/forcing/"
+        }
+    },
+    "time": {
+        "start_time": "2015-12-01 00:00:00",
+        "end_time": "2015-12-30 23:00:00",
+        "output_interval": 3600
+    },
+    "catchments": {
+        "cat-87": {
+            "formulations": [
+                {
+                    "name": "bmi_c",
+                    "params": {
```

Comment:
No, at least not right now. That would involve knowing them in advance (or at least I think it would), so we can't do that for everything. I think it makes more sense to not try that for a subset, though it could be considered.

Reply:
I wrote another comment in the formulation implementation class that is related here...I don't think they have to be known in advance, just checked for existence during formulation construction. But the question is answered, will resolve this and track the details on the implementation thread.

Reply:
Actually, I take it back. This at least originally wasn't going to be supported by the CFE example model (although I needed to change something later that may now enable it), but it _**is**_ supported by the formulation class ... sort of.  

If the right JSON key (`other_input_variables`) is in the formulation config, then the formulation (during its `create_formulation()` call) will pass the JSON value to the constructor of the `Bmi_C_Adapter` object when it creates its member instance of one.  The constructor will iterate through the valid input variable names for the model, and if it finds any matching keys in what got passed, it uses them to set those input variables.  At least in theory.

So I started to make this work, then abandoned it in the CFE example due to time constraints, then had to go back and change things again.  But I don't think I have tested this at all yet.  And at first glance, it looks like it won't quite handle full arrays of the variable values properly, so it probably won't work as expected.  I'll see if I can address that.